toolchain.mk

```diff
@@ -56,17 +56,37 @@ AARCH64_CROSS_COMPILE 		?= $(AARCH64_PATH)/bin/aarch64-linux-gnu-
 AARCH64_GCC_VERSION 		?= arm-gnu-toolchain-11.3.rel1-x86_64-aarch64-none-linux-gnu
 SRC_AARCH64_GCC 		?= https://developer.arm.com/-/media/Files/downloads/gnu/11.3.rel1/binrel/$(AARCH64_GCC_VERSION).tar.xz
 
+RUST_TOOLCHAIN_PATH 		?= $(TOOLCHAIN_ROOT)/rust
+
 .PHONY: toolchains
-toolchains: aarch32 aarch64
+toolchains: aarch32-toolchain aarch64-toolchain rust-toolchain
 
-.PHONY: aarch32
-aarch32:
+.PHONY: aarch32-toolchain
+aarch32-toolchain:
 	$(call dltc,$(AARCH32_PATH),$(SRC_AARCH32_GCC),$(AARCH32_GCC_VERSION))
 
-.PHONY: aarch64
-aarch64:
+.PHONY: aarch64-toolchain
+aarch64-toolchain:
 	$(call dltc,$(AARCH64_PATH),$(SRC_AARCH64_GCC),$(AARCH64_GCC_VERSION))
 
+# Download the Rust toolchain
+define dl-rust-toolchain
+	@if [ ! -d "$(1)" ]; then \
+		mkdir -p $(1); \
```

Comment:
Shouldn't the commands be separated with `&&` instead to propagate eventual errors?

Reply:
Done and along with that added some error handling too.